src/App.tsx

```diff
@@ -24,72 +24,145 @@ enum SortType {
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends React.Component <{}, ReorderOptions> {
+  state:ReorderOptions = {
+    sortType: SortType.NONE,
+    isReversed: false,
+    lightAlpha: false,
+    lightLenght: false,
+    lightreset: false,
+  };
+
+  sortAlpha = () => {
+    this.setState({
+      sortType: SortType.ALPHABET,
+      lightAlpha: true,
+      lightLenght: false,
+    });
+  };
+
+  sortLenght = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+      lightLenght: true,
+      lightAlpha: false,
+    });
+  };
+
+  sortRevers = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+      lightreset: !state.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+      lightAlpha: false,
+      lightLenght: false,
+      lightreset: false,
+    });
+  };
+
+  render() {
+    const visibleGoods = getReorderedGoods(goodsFromServer, this.state);
+
+    const showResetButton
+      = this.state.sortType !== SortType.NONE
+      || this.state.isReversed !== false;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={`button is-info 
+            ${this.state.lightAlpha ? '' : 'is-light'}`}
+            onClick={this.sortAlpha}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={`button is-success  
+            ${this.state.lightLenght ? '' : 'is-light'}`}
+            onClick={this.sortLenght}
+
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={`button is-warning 
+            ${this.state.lightreset ? '' : 'is-light'}`}
+            onClick={this.sortRevers}
+          >
+            Reverse
+          </button>
+
+          {showResetButton
+            ? (
+              <button
```

Comment:
You can use logical && instead of the ternary operator as you only decide whether to show the button or not
https://react.dev/learn/conditional-rendering